src/deployment.js

```diff
@@ -69,7 +71,10 @@ class Deployment {
     // Poll the deployment endpoint for status
     async check() {
       try {
-        const statusUrl = `https://api.github.com/repos/${this.repositoryNwo}/pages/deployment/status/${process.env['GITHUB_SHA']}`
+        const statusUrl = this.deploymentInfo != null ?
+          this.deploymentInfo["status_url"] :
+          `https://api.github.com/repos/${this.repositoryNwo}/pages/deployment/status/${process.env['GITHUB_SHA']}`
+        core.setOutput('page_url', this.deploymentInfo != null ? this.deploymentInfo["page_url"] : "");
```

Comment:
I think we also need to set an outputs section in the `action.yml`? https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#outputs

Reply:
ah, good catch! will adjust that as well :)

Reply:
updated action.yaml, and verified works thru
```
  deploy:
    runs-on: ubuntu-latest
    needs: build
    environment:
      name: github-pages-prod
      url: ${{ steps.deployment.outputs.page_url }}
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@expose-page-url
```